### ISSUE TITLE
Cannot remap keys in "extend-goto" mode

### ISSUE THREAD

**Issue by occivink** [mawww/kakoune#1299]
Or whatever the mode triggered by `G` is called.

Also the infobox could probably differentiate both goto and extend-goto.

**Comment by Delapouite** [comment]
> Also the infobox could probably differentiate both goto and extend-goto.

Yes, there could also be a difference between `v` and `V` in the info box. Something mentioning *locking* for `V`.

**Comment by Delapouite** [comment]
More generally there are also no easy ways to make distinct mapping for the various *flavors* of `object` mode: `<a-i>`, `<a-a>`, `[`, `{`, `]`, `}`, `[`, `<a-[>`, `<a-{>`, `<a-]>`, `<a-}>`…

Those are not really distinct mode, it's just the `SelectMode` switching from `Replace` to `Extend`, and `ObjectFlags` being a combination of `ToBegin`, `ToEnd`, `Inner`.

One solution could be to have access to this values in `%val{}` expansions. (i.e. `%val{select_mode}` and `%val{object_flags}`) Then it would be up to the user to implement custom logic in scripts accordingly.

**Comment by eraserhd** [comment]
This was resolved by #2736, right?
